src/store/reducer/buttonToggleSlice.ts

```diff
@@ -0,0 +1,78 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+export interface ButtonToggleState {
+  isRulerOn: boolean;
+  isCartOpen: boolean;
+  isPaintPopoverOpen: boolean;
+  isSavePopoverOpen: boolean;
+  isSideBarOpen: boolean;
+  isLoginModalOpen: boolean;
+}
+
+export const initialState: ButtonToggleState = {
+  isRulerOn: true,
+  isCartOpen: false,
+  isPaintPopoverOpen: false,
+  isSavePopoverOpen: false,
+  isSideBarOpen: false,
+  isLoginModalOpen: false,
+};
+
+export const ButtonToggleSlice = createSlice({
+  name: "ButtonToggle",
+  initialState,
+  reducers: {
+    switchRuler: (state: ButtonToggleState, action: PayloadAction<boolean>) => {
+      return {
+        ...state,
+        isRulerOn: action.payload,
+      };
+    },
+    switchCartDisplay: (state: ButtonToggleState) => {
+      return {
+        ...state,
+        isCartOpen: !state.isCartOpen,
+        isSideBarOpen: false,
+      };
+    },
+    usePaintBucket: (state: ButtonToggleState, action: PayloadAction<boolean>) => {
+      return {
+        ...state,
+        isPaintPopoverOpen: action.payload,
+        isSideBarOpen: false,
+      };
+    },
+    useSavePopover: (state: ButtonToggleState, action: PayloadAction<boolean>) => {
```

Comment:
The prefix "use" normally only uses in the name of hooks and these are redux dispatches. You can simply change the name of them to something like "switchSomething" or "toggleSomething" just an example. Ultimately, we want to keep all of them consistent.